go/coordinator/internal/utils/pulsar_admin.go

```diff
@@ -0,0 +1,44 @@
+package utils
+
+import (
+	"github.com/pingcap/log"
+	"go.uber.org/zap"
+
+	"github.com/apache/pulsar-client-go/pulsaradmin"
+	pulsar_utils "github.com/apache/pulsar-client-go/pulsaradmin/pkg/utils"
+)
+
+// This function creates topics in Pulsar. It takes in a list of topics and creates them in pulsar.
+// It assumes that the tenant and namespace already exist in Pulsar.
+func CreateTopics(pulsarWebServiceURL string, tenant string, namespace string, topics []string) error {
+	cfg := &pulsaradmin.Config{
+		WebServiceURL: pulsarWebServiceURL,
+	}
+	admin, err := pulsaradmin.NewClient(cfg)
+	if err != nil {
+		log.Error("Failed to create pulsar admin client", zap.Error(err))
+		return err
+	}
+
+	for _, topic := range topics {
+		topicSchema := "persistent://" + tenant + "/" + namespace + "/" + topic
+		topicName, err := pulsar_utils.GetTopicName(topicSchema)
+		if err != nil {
+			log.Error("Failed to get topic name", zap.Error(err))
+			return err
+		}
+		metadata, err := admin.Topics().GetMetadata(*topicName)
```

Comment:
Are you just using the metadata to check if it already exists? Can't you check for 409 on create?

Reply:
This will be a checking with a string, which is fragile.

Reply:
I don't agree that HTTP status codes are fragile but ok.